### ISSUE TITLE
Inspector hangs and crashes WindowServer

### ISSUE THREAD

**Issue by xTibor** [SerenityOS/serenity#962]
This is a kinda a two-in-one bug.

**Reproduction:**

1. Open Terminal
2. Run `ps` and check the process ID of `ps`
3. Run the Inspector with `ps` process ID + 1 (the ID what's going to be the Inspector's ID)
4. [Bug 1] Inspector hangs, cannot inspect itself.
5. Click and hold the left mouse cursor inside Inspector's frozen window and flail around for 10 seconds. Other methods that sends events to the window could also work.
6. [Bug 2] WindowServer crashes.

**Logs:**
```
WindowServer(12): Connection::post_message: Client buffer overflowed.
WindowServer(12): Connection{0x01542be6} (id=7, pid=20) misbehaved, disconnecting.
WindowServer(12) Couldn't find user region for V0xededee65
NP(error) fault at invalid address V0xededee65
WindowServer(12:12) Unrecoverable page fault, read from address 0xededee65
Note: Address 0xededee65 looks like it may be recently free()'d memory
CRASH: Page Fault. Process: WindowServer(12)
```

The stacktrace seems to vary, this is one of them:

```
0x08051568  WSWindowManager::process_mouse_event(WSMouseEvent&, WSWindow*&) +328
0x0013e6f8  Process::crash(int, unsigned int) +184
0x001178f8  illegal_instruction_handler +0
0x00117acf  page_fault_handler +405
0x00116c0a  page_fault_asm_entry +22
0x08051e0c  WSWindowManager::event(CEvent&) +48
0x08075a94  CObject::dispatch_event(CEvent&, CObject*) +86
0x08077bb8  CEventLoop::pump(CEventLoop::WaitMode) +502
0x08077d77  CEventLoop::exec() +71
0x080486be  main +754
0x080488c4  _start +94
```

**Comment by bugaevc** [comment]
I can't reproduce the WS crash; it must have been fixed at some point.

**Comment by IdanHo** [comment]
Fixed by 899698c31733cd0c4649c0e6ea7ee9059e25ae8b
